dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/icon/jdbc/JdbcCustomIconStore.java

```diff
@@ -137,4 +132,35 @@ public void update(CustomIcon customIcon) {
         customIcon.getLastUpdated(),
         customIcon.getKey());
   }
+
+  private String buildIconQuery(IconOperationParams iconOperationParams, String sql) {
+    SqlHelper hlp = new SqlHelper(true);
+
+    if (iconOperationParams.hasLastUpdated()) {
```

Comment:
@stian-sandvold or @enricocolasante did we want to move away from this style of generating SQL? Rather use MapSqlParameterSource and NamedParameterJdbcTemplate to have some SQL injection prevention.

Reply:
SQL injection risk is there if SQL contains input from the user. That is why validation is added here to make sure
user-entered input are valid.
I can change it to include parameterized input.

Reply:
Keywords are also user input. I am not sure there is no way to exploit this. My point is better safe than sorry by replacing an error prone practice with a safer one.